src/main/java/org/mule/extension/api/introspection/declaration/fluent/SourceDeclaration.java

```diff
@@ -59,4 +64,14 @@ public void setAttributesType(DataType attributesType)
     {
         this.attributesType = attributesType;
     }
+
+    public Optional<ExceptionEnricherFactory> getExceptionEnricherFactory()
+    {
+        return exceptionEnricherFactory;
+    }
+
+    public void setExceptionEnricherFactory(Optional<ExceptionEnricherFactory> exceptionEnricherFactory)
```

Comment:
same idea than before, I don't think we should force the optional in the interface and instead wrap it internally. We can have a default value of _Optional.empty()_, and then wrap whatever is passed on to the setter as an _Optional.ofNullable()_. That way, in the common case that the dev wants to actually set a valid factory, does not have to do an _Optional.of(myFactory)_